src/main/java/com/sngular/kloadgen/processor/objectcreatorfactory/impl/ProtobufObjectCreatorFactory.java

```diff
@@ -48,7 +47,16 @@ public ProtobufObjectCreatorFactory(final Object schema, final Object metadata)
     } else {
       throw new KLoadGenException("Unsupported schema type");
     }
-    this.metadata = (SchemaMetadata) metadata;
+
+    if (metadata instanceof SchemaMetadata) {
```

Comment:
It seems that this code is duplicated in the previous file.
Do it has sense to move it to a method in the interface to be inherit for both of them?